flink-connector-examples/src/main/java/io/pravega/example/flink/watermark/writer/FlinkWatermarkWriter.java

```diff
@@ -0,0 +1,67 @@
+package io.pravega.example.flink.watermark.writer;
+
+import io.pravega.client.stream.Stream;
+import io.pravega.connectors.flink.FlinkPravegaWriter;
+import io.pravega.connectors.flink.PravegaConfig;
+import io.pravega.connectors.flink.serialization.PravegaSerialization;
+import io.pravega.example.flink.Utils;
+import io.pravega.example.flink.watermark.Constants;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.streaming.api.TimeCharacteristic;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.IngestionTimeExtractor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
+ * Functionality
+ *     - Read data from a socket(127.0.0.1:9999) and write it into a Pravega stream(watermark-examples/output) with watermark.
+ */
+public class FlinkWatermarkWriter {
+    // Logger initialization
+    private static final Logger LOG = LoggerFactory.getLogger(FlinkWatermarkWriter.class);
+
+    public static void main(String[] args) throws Exception {
+        LOG.info("Starting Watermark Writer...");
+
+        // initialize the parameter utility tool in order to retrieve input parameters
+        ParameterTool params = ParameterTool.fromArgs(args);
+        PravegaConfig pravegaConfig = PravegaConfig
+                .fromParams(params)
+                .withDefaultScope(Constants.DEFAULT_SCOPE);
+
+        // create the Pravega input stream (if necessary)
+        Stream stream = Utils.createStream(
+                pravegaConfig,
+                params.get(Constants.STREAM_PARAM, Constants.OUTPUT_STREAM));
+
+        // initialize the Flink execution environment
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+        // Using event time characteristic
+        env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
+        env.getConfig().setAutoWatermarkInterval(2000);
+
+        // Get the data from socket and assign timestamp and a periodic watermark strategy onto the DataStream
+        DataStream<String> dataStream = env.socketTextStream(Constants.DEFAULT_HOST, Constants.DEFAULT_PORT)
+                .uid("Socket Stream")
+                .assignTimestampsAndWatermarks(new IngestionTimeExtractor<>())
+                .uid("Timestamps/Watermarks");
+
+        // Register a Flink Pravega writer with watermark enabled
+        FlinkPravegaWriter<String> writer = FlinkPravegaWriter.<String>builder()
+                .withPravegaConfig(pravegaConfig)
+                .forStream(stream)
+                .enableWatermark(true)
```

Comment:
I haven't seen the PR for this writer-side watermark functionality.  I would argue that `enableWatermark` should be renamed to `enableWatermarkPropagation`.  Also I don't understand how it would be implemented correctly.

Reply:
I think your suggestion for `enableWatermarkPropagation` is better. The implementation detail is in https://github.com/pravega/flink-connectors. I'm also struggling about how we can write a example to "show" that we ensure the functionality of the writer that watermarks are propagated into Pravega. We can only read `TimeWindow` from the stream which is NOT intuitive for presenting.

Reply:
As the release date coming nearer, should we have this `enableWatermarkingPropagation` change this `0.6.0` release or next `0.6.1`.

Reply:
Here's an idea: develop a job which computes an average of something over a certain time period, and writes the result to a stream.  Make the time period be configurable.   You can then apply the job recursively.  For example, a 1 minute aggregation job, writing to a 1 hour aggregation job, writing to a 1 day aggregation job.   

To show that it works, we need a healthy backlog of data, i.e. a few days worth.  The final job should produce correct output (which a job can validate).

Reply:
Can this be integrated with the current reader then comes into one case? I think it tested both the reader and writer.

Reply:
Yes, I think it could.  I like the simplicity of a recursive example.

Also consider implementing a handler for the 'late data' that the window function may emit.  Assuming that the ingestion watermarks are correct, one should never see any late data come from any of the recursive jobs.  This would be one form of self-validation.